accord-core/src/main/java/accord/local/AsyncCommandStores.java

```diff
@@ -1,92 +0,0 @@
-/*
```

Comment:
I think I’d meant to talk to you about this and never did. Some of the state tracking happening in C* was moved into Accord as explicit state changes in abstract base classes (ie: AbstractSafeCommandStore), and these changes were not compatible with the split sync/async class hierarchy. Since accord is meant to be async, I removed the sync class. I’m not sure this is necessary with the current state of the code. 

While we’re talking about it though, it seemed that these classes were meant to make sync and async operation a bit cleaner and remove some code duplication. Was there another goal that’s  less obvious?

Reply:
Yeah, the sync stores avoid any allocations or additional work in the normal case. Having both sync and async maintained ensures our API decisions are compatible with both approaches. Removing one is likely to lead to a situation where unpicking some dependency is much harder in future. Since the sync approach is potentially more efficient long term, I put quite a bit of work into ensuring both paradigms could be supported equally efficiently, so that we can more easily migrate in future.

This is the second time this work has been erased, so it would be nice to try and avoid that in future if possible. Perhaps we can add some commentary to this effect.

> Since accord is meant to be async

Not sure what you mean by that? Most of the `CommandStore` processing is synchronous, we currently have a thread per `CommandStore` but its work serving these requests is synchronous. I am far from certain that this will continue to be true indefinitely, so we shouldn't assume it I don't think. For data that is in cache, using the thread that happens to be running will likely be more efficient. Also, the separation of loading/writing concerns and thread in general in Cassandra is very likely to evolve. Virtual threads is a mid-term horizon that may very fundamentally alter what is optimal too.